### PR TITLE
GHAE support (on-hold)

### LINKED ISSUE #375: Add `--target-api-url` to `gh gei` commands
Right now for GHAE migrations, `gh gei` supports passing the `--target-api-url` to `migrate-repo` command to be able to make API calls against non default (`https://api.github.com`) bases addresses. The problem is that no other command in `gh gei` supports passing the `--target-api-url` in. We need to add the arg to all other `gh gei` commands and also make necessary changes to the `generate-script` command. 

## Todo
- [x] Add `--target-api-url` to all `gh gei` commands.
- [x] Modify `generate-script` to support passing the `--target-api-url` to `gh gei` commands.

### PR DESCRIPTION
**Pull request by ArinGhazarian**
Closes #375 

This PR adds the `--target-api-url` to all `gh gei` commands for GitHub to GitHub migration paths.

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)

### CODE REVIEW

File: src/gei/Commands/GenerateScriptCommand.cs [hunk 1]
```diff
@@ -112,6 +112,11 @@ public class GenerateScriptCommand : Command
             {
                 IsRequired = false
             };
+            var targetApiUrl = new Option<string>("--target-api-url")
+            {
+                IsRequired = false,
+                Description = "The URL of the target API, if not migrating to github.com (default: https://api.github.com). It will be ignored for ADO to GitHub migration path."
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateScriptCommand.cs:118]: Do I need to explicitly say that it will be ignored for ADO migration path?

Comment (dylan-smith) [reply src/gei/Commands/GenerateScriptCommand.cs:118]: I wouldn't, especially because we're about to remove the ADO migration path from this tool.

Comment (dylan-smith) [reply src/gei/Commands/GenerateScriptCommand.cs:118]: I would however mention that this is for GHAE

Comment (ArinGhazarian) [reply src/gei/Commands/GenerateScriptCommand.cs]: addressed in a02ad5bc93a104ef3c5b7ee19bdd1fe2df9836bd

File: RELEASENOTES.md [hunk 2]
```diff
@@ -1 +1 @@
-
+- Add `--target-api-url` to all `gh gei` commands for GitHub to GitHub migration paths.
```
Comment (dylan-smith) [inline RELEASENOTES.md:1]: I would mention that it is for GHAE support

File: src/Octoshift/GithubClient.cs [hunk 3]
```diff
@@ -115,6 +115,11 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url)
         {
             url = url?.Replace(" ", "%20");
 
+            if (httpMethod == HttpMethod.Post && url is not null && url.Trim().EndsWith("/v3/graphql", StringComparison.OrdinalIgnoreCase))
+            {
+                url = url.Replace("/v3/graphql", "/graphql", StringComparison.OrdinalIgnoreCase);
+            }
```
Comment (dylan-smith) [inline src/Octoshift/GithubClient.cs:121]: this turned out not as ugly as I was thinking. This ain't so bad!

Comment (ArinGhazarian) [reply src/Octoshift/GithubClient.cs:121]: Yeah I guess this is the easiest approach to put the logic in the client instead of the `GithubApi` 😃

**Comment by dylan-smith** [comment]
Closing this as we're not likely to revive it at this point.

**Pull request closed without merging**